### PR TITLE
Feature/support forkjoinpool

### PR DESCRIPTION
**Pull request by xu1009**
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [ ] Update the documentation to include this new feature.
- [ ] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. https://github.com/apache/skywalking/issues/10680
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
This doc should be updated, https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/bootstrap-plugins/

And as a new plugin is being added, you should update `supported-list` and `plugin-list`, otherwise, CI would fail.

**Comment by xu1009** [comment]
ok

File: apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/ForkJoinWorkerQueueMethodInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import java.lang.reflect.Method;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.ContextSnapshot;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+public class ForkJoinWorkerQueueMethodInterceptor implements InstanceMethodsAroundInterceptor {
```
[the remark this answers is not in the document]
Comment (xu1009) [reply apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/ForkJoinWorkerQueueMethodInterceptor.java:30]: not get your mean，i have checked InstanceMethodsAroundInterceptorV2， it seems not suitable in current

[the remark this answers is not in the document]
Comment (wu-sheng) [reply apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/ForkJoinWorkerQueueMethodInterceptor.java:30]: You created one span in before, and stop in the after. The STOP is actually calling active span from threadlocal.
A better way is, you put the created span in the MethodInvocationContext, and read from it in the after.

File: apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinWorkerQueueInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.v2.ClassEnhancePluginDefineV2;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.v2.InstanceMethodsInterceptV2Point;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.v2.StaticMethodsInterceptV2Point;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.NameMatch;
+
+public class ForkJoinWorkerQueueInstrumentation extends ClassEnhancePluginDefineV2 {
```
Comment (wu-sheng) [inline apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinWorkerQueueInstrumentation.java:32]:
```suggestion
public class ForkJoinWorkerQueueInstrumentation extends ClassInstanceMethodsEnhancePluginDefineV2 {
```

File: apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinTaskInstrumentation.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.StaticMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.HierarchyMatch;
+
+public class ForkJoinTaskInstrumentation extends ClassEnhancePluginDefine {
```
Comment (wu-sheng) [inline apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinTaskInstrumentation.java:32]:
```suggestion
public class ForkJoinTaskInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
```

File: apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinTaskInstrumentation.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Override
+    public StaticMethodsInterceptPoint[] getStaticMethodsInterceptPoints() {
+        return new StaticMethodsInterceptPoint[0];
+    }
+
+    @Override
+    public boolean isBootstrapInstrumentation() {
+        return true;
+    }
```
Comment (wu-sheng) [inline apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinTaskInstrumentation.java:72]:
```suggestion
```

I think we don't need this once we inherit from the `ClassInstanceMethodsEnhancePluginDefine`.

File: apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinWorkerQueueInstrumentation.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Override
+    public StaticMethodsInterceptV2Point[] getStaticMethodsInterceptV2Points() {
+        return new StaticMethodsInterceptV2Point[0];
+    }
+
+    @Override
+    public boolean isBootstrapInstrumentation() {
+        return true;
+    }
```
Comment (wu-sheng) [inline apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinWorkerQueueInstrumentation.java:79]:
```suggestion
```

I think we don't need this once we inherit from the `ClassInstanceMethodsEnhancePluginDefineV2`.

**Comment by wu-sheng** [comment]
Notice, the code format should be fixed.

```
  Error:  /home/runner/work/skywalking-java/skywalking-java/test/plugin/scenarios/jdk-forkjoinpool-scenario/src/main/java/org/apache/skywalking/apm/testcase/jdk/forkjoinpool/controller/CaseController.java:44:5: 'VARIABLE_DEF' has more than 1 empty lines before. [EmptyLineSeparator]
  Audit done.
  [INFO] There is 1 error reported by Checkstyle 8.38 with /home/runner/work/skywalking-java/skywalking-java/apm-checkstyle/checkStyle.xml ruleset.
  Error:  scenarios/jdk-forkjoinpool-scenario/src/main/java/org/apache/skywalking/apm/testcase/jdk/forkjoinpool/controller/CaseController.java:[44,5] (whitespace) EmptyLineSeparator: 'VARIABLE_DEF' has more than 1 empty lines before.
```

**Comment by xu1009** [comment]
> 

not report this in my loacl when execute the test, it seems not check test code style

File: apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinTaskInstrumentation.java (hunk 3 above)
Comment (wu-sheng) [reply apm-sniffer/bootstrap-plugins/jdk-forkjoinpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/forkjoinpool/define/ForkJoinTaskInstrumentation.java:32]: @xu1009 I think you missed this too.

**Pull request merged**